ArchiSteamFarm/Plugins/PluginsCore.cs

```diff
@@ -655,6 +699,102 @@ public static class PluginsCore {
 		}
 	}
 
+	[UnconditionalSuppressMessage("AssemblyLoadTrimming", "IL3000", Justification = "We don't care about trimmed assemblies, as we need it to work only with the known (used) ones")]
+	internal static async Task<bool> UpdatePlugins(Version asfVersion, GlobalConfig.EUpdateChannel? updateChannel) {
+		ArgumentNullException.ThrowIfNull(asfVersion);
+
+		if (updateChannel.HasValue && !Enum.IsDefined(updateChannel.Value)) {
+			throw new InvalidEnumArgumentException(nameof(updateChannel), (int) updateChannel, typeof(GlobalConfig.EUpdateChannel));
+		}
+
+		if (ActivePlugins.Count == 0) {
+			return false;
+		}
+
+		if (ASF.WebBrowser == null) {
+			throw new InvalidOperationException(nameof(ASF.WebBrowser));
+		}
+
+		updateChannel ??= ASF.GlobalConfig?.UpdateChannel ?? GlobalConfig.DefaultUpdateChannel;
+
+		bool restartNeeded = false;
+
+		ASF.ArchiLogger.LogGenericInfo(Strings.PluginUpdatesChecking);
+
+		// We update plugins one-by-one to limit memory pressure from potentially big release assets
```

Comment:
While I'm still in favour of this approach, I still have to mention the existance of the `OptimizationMode` global setting that could be used here for the sake of completeness.

Reply:
We could try to make it run in parallel since the assumption is that paths won't clash with each other, although it's kinda risky - I need some neat solution here to ensure we don't have a situation where plugin2 is in subdirectory of plugin1 and both are updated at once.

Reply:
By the way, we can probably avoid loading binaries into the memory altogether and only use `Stream` but that would make implementation of ProgressReporter either difficult or impossible

Reply:
We'd need to rewrite update mechanism to extract to some new directory this way, that's possible but definitely out of the scope of this PR, since current mechanism will corrupt files if you kill it during update.

Something to perhaps consider for the future.